include/h2o/memory.h

```diff
@@ -301,6 +305,13 @@ void h2o_append_to_null_terminated_list(void ***list, void *element);
 
 /* inline defs */
 
+inline void *h2o_memcpy(void *dst, const void *src, size_t n)
+{
+    if (src)
```

Comment:
@deweerdt 
> Maybe we could have an assert that if `src == NULL`, then `n` must be `0`? I believe this would cover both cases.

Generally speaking, I do not think there's a _need_ to ensure UB when we have a logic flaw (we would have UB everywhere in such case), but I have no objection taking the approach for memcpy.